File: C++/first-element-with-unique-frequency.cpp
// Time:  O(n)
// Space: O(n)

// freq table
class Solution {
public:
    int firstUniqueFreq(vector<int>& nums) {
        unordered_map<int, int> cnt;
        for (const auto & x : nums) {
            ++cnt[x];
        }
        unordered_map<int, int> cnt2;
        for (const auto& [_, v] : cnt) {
            ++cnt2[v];
        }
        for (const auto& x : nums) {
            if (cnt2[cnt[x]] == 1) {
                return x;
            }
        }
        return -1;
    }
};

File: Python/first-element-with-unique-frequency.py
# Time:  O(n)
# Space: O(n)

import collections


# freq table
class Solution(object):
    def firstUniqueFreq(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cnt = collections.defaultdict(int)
        for x in nums:
            cnt[x] += 1
        cnt2 = collections.defaultdict(int)
        for v in cnt.itervalues():
            cnt2[v] += 1
        return next((x for x in nums if cnt2[cnt[x]] == 1), -1)
